packages/tailwindcss/src/compile.ts

```diff
@@ -31,9 +31,9 @@ export function compileCandidates(
   }
 
   // Sort the variants
-  let variants = designSystem.getUsedVariants().sort((a, z) => {
-    return designSystem.variants.compare(a, z)
-  })
+  let variants = sortAndGroup(designSystem.getUsedVariants(), (a, z) =>
+    designSystem.variants.compare(a, z),
+  )
```

Comment:
I think we should move this sorting into `designSystem.getUsedVariants()`

Maybe the name should be something like `getUsedVariantGroups()` or something then as well?